include/xeus-zmq/xserver_zmq.hpp

```diff
@@ -63,6 +71,15 @@ namespace xeus
     std::unique_ptr<xserver> make_xserver_shell_main(xcontext& context,
                                                      const xconfiguration& config,
                                                      nl::json::error_handler_t eh = nl::json::error_handler_t::strict);
+
+    XEUS_ZMQ_API
+    std::unique_ptr<xserver> make_xserver_uv_shell_main(
```

Comment:
I think adding a comment showing how this function should be used could be helpful. The `xeus::xkernel` constructor accepts a function that takes 3 arguments only, so on would probably have to use a lamba capturing the loop and the hook before calling this function.